pkg/api/core/shoot/warnings.go

```diff
@@ -42,6 +44,14 @@ func GetWarnings(_ context.Context, shoot, oldShoot *core.Shoot, credentialsRota
 	if oldShoot != nil {
 		warnings = append(warnings, getWarningsForDueCredentialsRotations(shoot, credentialsRotationInterval)...)
 		warnings = append(warnings, getWarningsForIncompleteCredentialsRotation(shoot, credentialsRotationInterval)...)
+
+		if versionutils.ConstraintK8sLess125.Check(semver.MustParse(shoot.Spec.Kubernetes.Version)) &&
+			versionutils.ConstraintK8sGreaterEqual123.Check(semver.MustParse(shoot.Spec.Kubernetes.Version)) {
```

Comment:
```suggestion
if shootVersion := semver.MustParse(shoot.Spec.Kubernetes.Version); versionutils.ConstraintK8sLess125.Check(shootVersion) && versionutils.ConstraintK8sGreaterEqual123.Check(shootVersion) {
```

However, using `MustParse` is dangerous, let's maybe not use it?